### DISCUSSION TITLE
How to get the hash param in Next 13.5.5 Pages Router

### DISCUSSION THREAD

**Discussion by Judge-Paul** [vercel/next.js#61112] [Help]
### Summary

I am on Next 13.5.5 Pages Router 
I need to get what the current hash is so I can conditionally render something

I want to do something like
if (hash === "section1") {
  console.log("You are in section1")
} else {
  console.log("You are in section2")
}

This isn't my exact usecase but if I can get this to work I have solved my problem

**Comment by icyJoseph** [comment]
Hi,

Before I give you some code, there's one thing to remember here, the `hash` or `fragment` of a URL, that is, the `#` and its following characters, are never sent to the server. The browser keeps them. You can do an search online to read about why this happens.

- https://developer.mozilla.org/en-US/docs/Web/API/URL/hash
- https://github.com/vercel/next.js/issues/55044#issuecomment-1708001099

Alright, so if we think of client side only...

You can synchronize with the hash using the https://developer.mozilla.org/en-US/docs/Web/API/Window/hashchange_event:

```js
import { useEffect, useState } from 'react';

const useCurrentHash = () => {
   const [value, setValue] = useState("") // make this SSR resilient by not using the window here

   useEffect(() => {
       const handler = () => setValue(window.location.hash);
       window.addEventListener("hashchange", handler);
       return () => {
            window.removeEventListener("hashchange", handler);
       }
   }, []);

   return value
}
```

And that should give you a good approximation. Note that the first frame starts with `""`

**Reply by Judge-Paul** [reply]
Thanks for your reply 

I have already solved this using 
``` javascript
const router = useRouter();

  useEffect(() => {
    const hash = router.asPath.split("#")[1];
    setActive(hash);
  }, [router.asPath]);
```
